### ISSUE TITLE
Why not add an issuetitleaddon field ?

### ISSUE THREAD

**Issue by jeromevillebrun** [plk/biblatex#996]
Hi, 
For a bibliographic style I am writing, I would need a field that go just after the `issuetitle` of entries which use the `issue` macro. There is already a `titleaddon` field, so why wouldn't there be a `issuetitleaddon` ? It would be very usefull for many french bibliography styles (and maybe other languages). 

In french, a typical example of what would go in this field would be an inscription like "numéro spécial" or "dossier thématique" (I don't know the translation in english, but it doesn't matter since `issuetitleaddon` would be a litteral field, exactly as `titleaddon` is).

If prefered, I can try to do a Pull Request, but for the moment, I have no experience in creating new fields (and I also don't where the documentation is).

**Comment by moewew** [comment]
You can 'easily' add new fields yourself with a new datamodel file. See https://tex.stackexchange.com/q/163303/35864.

To get a `issuetitleaddon` you could say
```latex
\documentclass[british]{article}
\usepackage[T1]{fontenc}
\usepackage[utf8]{inputenc}
\usepackage{babel}
\usepackage{csquotes}

\begin{filecontents}{issuetitleaddon.dbx}
\ProvidesFile{issuetitleaddon.dbx}[2020/04/14 issuetitleaddon]

\DeclareDatamodelFields[type=field,datatype=literal]{issuetitleaddon}
\DeclareDatamodelEntryfields[article,periodical]{issuetitleaddon}
\end{filecontents}

\usepackage[style=authoryear, backend=biber, datamodel=issuetitleaddon]{biblatex}

\renewbibmacro*{issue}{%
  \ifboolexpr{
    test {\iffieldundef{issuetitle}}
    and
    test {\iffieldundef{issuesubtitle}}
  }
    {}
    {\printtext[issuetitle]{%
       \printfield[titlecase]{issuetitle}%
       \setunit{\subtitlepunct}%
       \printfield[titlecase]{issuesubtitle}}}%
  \printfield{issuetitleaddon}}

\begin{filecontents}{\jobname.bib}
@article{appleby,
  author          = {Humphrey Appleby},
  title           = {On the Importance of the Civil Service},
  date            = {1980},
  journal         = {Journal of the Civil Service},
  volume          = {234},
  issuetitle      = {Importance Issue},
  issuetitleaddon = {Special Issue 123},
}
\end{filecontents}
\addbibresource{\jobname.bib}
\addbibresource{biblatex-examples.bib}


\begin{document}
\cite{sigfridsson,appleby}
\printbibliography
\end{document}
```

Note that `filecontents` is just used to make the example file self-contained. In your real world example you would save
```latex
\ProvidesFile{issuetitleaddon.dbx}[2020/04/14 issuetitleaddon]

\DeclareDatamodelFields[type=field,datatype=literal]{issuetitleaddon}
\DeclareDatamodelEntryfields[article,periodical]{issuetitleaddon}
```
as `issuetitleaddon.dbx` and would put it in a place where LaTeX can find it (for example the same place as your main `.tex` file). Similarly for your `.bib` file.

If you are writing a custom style called `<mysytle>` with `<mysytle>.bbx` and `<mysytle>.cbx` already, then you would call the `.dbx` file `<mystyle.dbx>` and there wouldn't be a need to manually load it with `datamodel=<mystyle>,` as it will automatically be loaded when you say `style=<mystyle>,`.

---

I'm not sure if `issuetitleaddon` needs to be added to the standard data model. The standard data model can't cover every last possible use case people may have. It is intended to be a solid base for most uses, but there was always the intention that users can extend it (via `.dbx` files, as shown above) if they need new fields or new entry types (https://tex.stackexchange.com/q/175776/35864). I usually try to be conservative and avoid adding too many new fields for very narrow use cases (see for example #971, #969, #908, #880, #837, ...). You do have a point that `journaltitleaddon` would be analogue to `titleaddon`, `booktitleaddon` and `maintitleaddon`. But then again we don't have `journaltitleaddon` either. `issuetitleaddon` and `journaltitleaddon` would be very specialised indeed.

Note that usually the subdivision of a journal `volume` is `number` and that's where I would normally put even things like `Suppl. 1` (see https://github.com/plk/biblatex/issues/726), so semantically that appears to be what I think your `issuetitleaddon` should do here (if I understand correctly). (I'm not claiming that `number` gives the best output here in the standard styles without modifications, just that semantically it could feel right here.)

**Comment by jeromevillebrun** [comment]
Thank you very much for your answer : it solved my problem. 
Of course, it is up to you to choose whether this field has to be added or not (and close this issue). I understand that since it is so easy to create new fields, adding it to the package may be useless.

**Comment by moewew** [comment]
Hrmm, this is a tough one. Usually I'd say we don't want to add new fields with a very narrow use case to the default data model (since people can add them if they need them), but since

* we already have `titleaddon`, `booktitleaddon` and `maintitleaddon` (so it feels like morally, `issuetitleaddon` and `journaltitleaddon` could have been there from the start)
* the semantic for `issuetitleaddon` and `journaltitleaddon` would be clear and
* it is easy to integrate these fields into the existing drivers with minimal backwards compatibility impact,

I'm much more inclined than usual to say that we might as well go for it.

@plk What do you think?

**Comment by moewew** [comment]
Suggested changes to the standard styles are at #1000. Comments are welcome.

**Comment by moewew** [comment]
The `issuetitleaddon`  and `journaltitleaddon` fields are available in `biblatex` 3.15, which was released a few days back and is now available in MikTeX and TeX live.
